src/sqlfluff/core/parser/segments/ephemeral.py

```diff
@@ -72,7 +72,7 @@ def wrapped_match_method(self, segments: tuple, parse_context):
             # We shouldn't allow nested ephemerals. If they're present, don't create
             # another. This can happen when grammars call super() on their match method.
             if len(segments) == 1 and segments[0].is_type("ephemeral"):
-                return MatchResult.from_matched(segments)
+                return MatchResult.from_matched(segments)  # pragma: no cover
```

Comment:
Good question. I don't think it can be executed anymore based on the current dialects, but it's possible that _maybe_ it could be if someone did something a bit unusual. I'm tempted to keep it but I might add a warning with it.